src/go/rpk/pkg/config/config.go

```diff
@@ -223,89 +145,52 @@ func AvailableModes() []string {
 	}
 }
 
-func Check(conf *Config) (bool, []error) {
-	configMap, err := toMap(conf)
-	if err != nil {
-		return false, []error{err}
-	}
-
-	v := viper.New()
-	err = v.MergeConfigMap(configMap)
-	if err != nil {
-		return false, []error{err}
-	}
-	return check(v)
-}
-
-func check(v *viper.Viper) (bool, []error) {
-	errs := checkRedpandaConfig(v)
+func Check(cfg *Config) (bool, []error) {
```

Comment:
Where is this used?

Reply:
saw -- this should potentially be a method on the Config itself, `func (*Config) Validate() []error` ?